python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -25,6 +30,22 @@
 regex = re.compile('[*?\[\]]')
 logger = getLogger()
 
+def escapeWhiteSpace(somePath):
+    """
+    This function allows us to use whitespaces better in file paths.
+    1) Yes we should be using something out of Python standard libs but none come to mind which aren't deprecated/undocumented
+    2) Yes this can be removed by using subprocess.call, but this is invasive and requires rewriting even more of this class than is needed (and script injections!)
+    3) Escaping the whitespace to be compatible with bash is probably the sanest thing we can do for all strings which allows us to support whitespace in filenames
+    4) This does NOT perform any level of security/safety check and I'm sure a user could inject 'rm -fr' commands into their job description\
+       but other security practices should isolate their choices from other users.
+    Args:
+        somePath (str): somePath which may contain whitespaces
+    """
+    returnable = str(somePath)
+    for i in (' ', ';', '(', ')',):
+        returnable = returnable.replace(i, "\\"+i)
+    return returnable
+
```

Comment:
@drmarkwslater I was concerned about strange differences in behavior between the instance we run here and the code run on the WN in this case.
I'm happy to change to this (if I can find an example) I just generally avoid things with big deprecation warnings but I'm not tied to this implementation by any means.